### PR TITLE
Use unified DB-API in codebase

### LINKED ISSUE #2781: Unify db api
**Is your feature request related to a problem? Please describe.**
Current db.py make so many special check and branch to adapt different DB engines. We need to unify the DB api interface according to this [doc](https://www.python.org/dev/peps/pep-0249/). Especially for maxcompute.

**Describe the solution you'd like**
1. Modify maxcompute.py to adapt db api
2. Provide a query interface in the db.py

**Describe alternatives (optional)**

**Additional Notes**

### PR DESCRIPTION
**Pull request by lhw362950217**
Fixes #2781

### CODE REVIEW

File: python/runtime/xgboost/predict.py [hunk 1]
```diff
@@ -124,19 +121,8 @@ def predict_and_store_result(bst, dpred, feature_file_id, model_params,
...
-                               conn,
-                               result_table,
-                               result_column_names,
-                               100,
-                               hdfs_namenode_addr=hdfs_namenode_addr,
-                               hive_location=hive_location,
-                               hdfs_user=hdfs_user,
-                               hdfs_pass=hdfs_pass) as w:
+    with db.buffered_db_writer(conn, result_table, result_column_names,
+                               100) as w:
```
Comment (typhoonzero) [inline python/runtime/xgboost/predict.py:125]: Why remove the other parameters?

File: python/runtime/pai/submitter.py [hunk 2]
```diff
@@ -130,15 +132,18 @@ def get_oss_model_url(model_full_path):
     return "oss://%s/%s" % (oss.SQLFLOW_MODELS_BUCKET, model_full_path)
 
 
+def parse_maxcompute_dsn(datasource):
```
Comment (sneaxiy) [inline python/runtime/pai/submitter.py:135]: Do we need this method or just write `MaxComputeConnection.get_uri_parts(datasource)` when calling?

**Review by sneaxiy — APPROVED** [review]
LGTM basically.

File: python/runtime/xgboost/predict.py (hunk 1 above)
Comment (lhw362950217) [reply python/runtime/xgboost/predict.py:125]: Parameters like `hdfs_pass` can be passed by connection uri, they are driver specific. We have a unified way to save these params (in `conn.params dict`), we can get them  by `conn.param(param_name)` when needed.

File: python/runtime/pai/submitter.py (hunk 2 above)
Comment (lhw362950217) [reply python/runtime/pai/submitter.py:135]: Good, idea. I will figure this out and do some modify if possible. In next PR :)

File: python/runtime/xgboost/predict.py (hunk 1 above)
Comment (typhoonzero) [reply python/runtime/xgboost/predict.py:125]: I see, the `db.py` file was folded and I missed the file.

**Pull request merged**